Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM python:3.9-slim-buster as base
+
+EXPOSE 5050
+WORKDIR /todo-app
+COPY . /todo-app
+RUN pip install "poetry==1.1.10" \
+&& poetry config virtualenvs.create false --local \
+&& poetry install 
+
+FROM base as development
+ENTRYPOINT poetry run flask run -h 0.0.0.0 -p 5050
+
+FROM base as production
+RUN poetry add gunicorn
```

Comment:
Rather than adding gunicorn to your dependencies within the production image, no real harm in just adding it permanently outside of Docker. Then you can manage the version in source control (via pyproject.toml) alongside your other dependencies.